aptos-move/framework/aptos-framework/sources/stake.move

```diff
@@ -843,94 +840,72 @@ module aptos_framework::stake {
     /// pending inactive validators so they no longer can vote.
     /// 4. The validator's voting power in the validator set is updated to be the corresponding staking pool's voting
     /// power.
-    public(friend) fun on_new_epoch() acquires StakePool, AptosCoinCapabilities, ValidatorConfig, ValidatorPerformance, ValidatorSet {
+    public(friend) fun on_new_epoch() acquires StakePool, AptosCoinCapabilities, ValidatorConfig, ValidatorSet {
         let validator_set = borrow_global_mut<ValidatorSet>(@aptos_framework);
         let config = staking_config::get();
-        let validator_perf = borrow_global_mut<ValidatorPerformance>(@aptos_framework);
 
-        // Process pending stake and distribute rewards for each currently active validator.
+        // Update stake pool with processing pending stake and distributing rewards.
         let i = 0;
         let len = vector::length(&validator_set.active_validators);
         while (i < len) {
             let validator = vector::borrow(&validator_set.active_validators, i);
-            update_stake_pool(validator, validator_perf, validator.addr, &config);
-            i = i + 1;
-        };
-
-        // Process pending stake and distribute rewards for each currently pending_inactive validator (requested to
-        // leave but not removed yet).
-        let i = 0;
-        let len = vector::length(&validator_set.pending_inactive);
-        while (i < len) {
-            let validator = vector::borrow(&validator_set.pending_inactive, i);
-            update_stake_pool(validator, validator_perf, validator.addr, &config);
-            i = i + 1;
+            update_stake_pool(validator, &config);
+            // Remove the validator from the set if it is pending to leave.
+            if (validator.pending_inactive) {
+                let _ = vector::swap_remove(&mut validator_set.active_validators, i);
```

Comment:
nit: Remove "let _ =" if the returned value is not used anyway.

Reply:
done.